File: typescript/resistor-color-duo/resistor-color-duo.ts
const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
]

const colorCode = (color: string): number => COLORS.indexOf(color)

export const decodedValue = ([tens, ones]: string[]): number => {
  return colorCode(tens) * 10 + colorCode(ones)
}

File: ruby/resistor-color-duo/resistor_color_duo.rb
module ResistorColorDuo
  COLOR_CODES = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }.freeze

  def self.value(colors)
    colors[0..1]
      .map { |c| COLOR_CODES[c.to_sym] }
      .join('')
      .to_i
  end
end
